src/gluonts/mx/trainer/_base.py

```diff
@@ -272,12 +275,15 @@ def loop(
                     ):
                         self.avg_strategy.load_averaged_model(net)
 
-                    with tqdm(
-                        itertools.islice(
-                            batch_iter, self.num_batches_per_epoch
-                        ),
-                        total=self.num_batches_per_epoch,
-                    ) as it:
+                    batch_iter_length = None
+
+                    if num_batches_to_use:
+                        batch_iter = itertools.islice(
```

Comment:
Can't this be unconditional, since slicing `None` is the same as not slicing at all?